lib/scaffold.dart

```diff
@@ -424,8 +420,8 @@ class BackdropScaffoldState extends State<BackdropScaffold>
 
   @override
   void dispose() {
+    _animationController?.dispose();
```

Comment:
The bug is not with `_shouldDisposeAnimationController`, it is calling of `_animationController?.dispose();` before `super.dispose` here, that has fixed the issue.

I suggest reverting back the `_shouldDisposeAnimationController` logic if the above is correct.